### ISSUE TITLE
Chunk header usage in C

### ISSUE THREAD

**Issue by MatthiasKillat** [eclipse-iceoryx/iceoryx#585]
## Brief feature description

The current chunk header is a C++ class and cannot be accessed safely purely with C.
We need an API (or C struct) to allow using the chunk header in C.

## Detailed information

The detailed approach needs to be discussed and depends on the desired level of functionality and configuration options the header shall provide to the user.

**Comment by neilisaac** [comment]
For my own purposes it is sufficient to expose the size related functions: https://github.com/eclipse-iceoryx/iceoryx/compare/master...neilisaac:c-chunk-header-size

Would you like a pull request for this change?

**Comment by elBoberido** [comment]
@neilisaac of course. I had a brief look. You need to add some tests and I guess @MatthiasKillat knows more about the requirements for the C-API than me.

**Comment by MatthiasKillat** [comment]
@neilisaac @elBoberido The additions look fine (no full review) and do not break any existing API. We could even use them in CycloneDDS as well (where, due to API freeze a while ago I  partially created some of them for convenience).

I am fine with having a pull request (and reviewing it), but how about tests? We would need some tests of the C-API even if it is a thin wrapper. Do you agree @elBoberido? We can also add them in a separate PR (but all in one PR would be better I guess, as it is not too large).

**Comment by elBoberido** [comment]
@neilisaac you can look at `test_request_header.cpp` from this PR https://github.com/eclipse-iceoryx/iceoryx/pull/1161/files for the tests. They should more or less be like `getChunkHeaderWorks`

**Comment by neilisaac** [comment]
Sorry I missed the comments.  I'd be happy to add tests and make a proper PR as time permits.

**Comment by elBoberido** [comment]
@neilisaac no problem. We are looking forward to your contribution and hopefully the next release is not that far away.
